javagen/src/main/java/com/azure/autorest/util/SchemaUtil.java

```diff
@@ -90,13 +89,17 @@ public static String getDiscriminatorSerializedName(ObjectSchema compositeType)
         if (compositeType.getDiscriminator() != null) {
             discriminator = compositeType.getDiscriminator().getProperty().getSerializedName();
         } else if (compositeType.getDiscriminatorValue() != null) {
-            for (ComplexSchema parent : compositeType.getParents().getAll()) {
-                if (((ObjectSchema) parent).getDiscriminator() != null) {
+            for (Schema parent : compositeType.getParents().getAll()) {
+                if (parent instanceof ObjectSchema && ((ObjectSchema) parent).getDiscriminator() != null) {
                     discriminator = ((ObjectSchema) parent).getDiscriminator().getProperty().getSerializedName();
                     break;
                 }
             }
         }
+        if (discriminator == null) {
+            throw new IllegalStateException(String.format("discriminator not found in type %s and its parents",
+                    compositeType.getLanguage().getJava().getName()));
+        }
```

Comment:
This appear to be m4 problem. Spec put the discriminator property to a parent class.
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/externalSecuritySolutions.json#L180-L199